### PR TITLE
904 Adding the "Open with..." button

### PR DESCRIPTION
**Pull request by shawncrawley**
The styling that I applied may not be the best option. It works, but there may be a better (or more in-line with hydroshare) way of doing it. Will someone qualified to make that call take a look at this?

### CODE REVIEW

**Comment by hydrobuild** [comment]
Can one of the admins verify this patch?

**Comment by horsburgh** [comment]
@mauriel133 - can you have a look at the styling and make sure it is consistent with the rest of the resource landing page?

**Comment by mauriel133** [comment]
Sure, is this branch up to date with the migration files issue? If so I can deploy it in a VM and test it.

**Comment by shawncrawley** [comment]
@mauriel133 It should be. I just merged with the latest dev about 15
minutes ago right before making the pull request.

**Comment by mjstealey** [comment]
> is this branch up to date with the migration files issue?

Check the `git log`, if you find this commit, you are good:
```bash
commit b6dd3f89551c1652cb055ee898ca8e37ea8029c6
Merge: cc044a4 a2a2144
Author: Michael J. Stealey <michael.j.stealey@gmail.com>
Date:   Mon Jan 25 15:48:28 2016 -0500

    Merge pull request #906 from hydroshare/892-migration-fix

    892 migration fix
```

**Comment by shawncrawley** [comment]
@mauriel133 @mjstealey Yeah, the log confirms it. You are good to go.

**Comment by mauriel133** [comment]
@shawncrawley, could you create a resource on  https://riverdev.hydroshare.org/ that I can test with?

**Comment by shawncrawley** [comment]
@mauriel133 I wasn't sure what you wanted, since this feature requires a Web App resource that is associated with other resource types. Thus, I created a Generic resource, a NetCDF resource, and a Web App resource that is associated with Generic and NetCDF resources. Here are the links to them:

https://riverdev.hydroshare.org/resource/d046296a45f940aab9c24ace07aa5baa/
https://riverdev.hydroshare.org/resource/5139ed6f97924b7caa1e58905756b629/
https://riverdev.hydroshare.org/resource/6dc0bc7b03914c15adb5a73d238bf0b4/

File: hs_tools_resource/static/css/toolresource.css [hunk 1]
```diff
@@ -15,4 +15,20 @@
...
     width: 100px;
     height: 100px;
-}
\ No newline at end of file
+}
+.dropdown-webapp-icon {
+    width: 40px;
+    height: 40px;
+    border: 0;
+    border-radius: 4px;
```
Comment (mauriel133) [inline hs_tools_resource/static/css/toolresource.css]: You specify a border radius, but no border to use it.

**Comment by shawncrawley** [comment]
@mauriel133 I just realized that the code at the heart of this change didn't actually get pushed. I think I accidently stashed it right before pushing. So I just did another commit/push that should be up-to-date.

**Comment by mauriel133** [comment]
@shawncrawley, let me know when it's been updated and I'll redeploy.

**Comment by shawncrawley** [comment]
@mauriel133 The branch has been merged with dev, updated, and tested. It should be good to go now.

**Comment by pkdash** [comment]
@mauriel133 Please see if it is ready for +1

File: hs_tools_resource/static/js/toolresource.js [hunk 2]
```diff
@@ -22,4 +22,25 @@ $(document).ready(function() {
...
+            .addClass(style + 'default-webapp-icon');
+    };
+    $('.user-webapp-icon, .dropdown-user-webapp-icon').each(function() {
+        if ($(this).attr('src') == '') {
+            setDefaultIcon($(this));
+        }
+    }).on('error', function() {
+        setDefaultIcon($(this));
+    });
+    $('h2').attr('style', 'width: calc(100% - 110px);');
```
Comment (mauriel133) [inline hs_tools_resource/static/js/toolresource.js]: This selector is too broad and can create problems in the future. I suggest to add a class to the h2 title tag and use it in this selector.

The selector would then look like this:
> $('h2.resource-title').attr('style', 'width: calc(100% - 110px);');

**Comment by mauriel133** [comment]
@shawncrawley, it looks great!

I just made one minor suggestion. Let me know what you think.

File: hs_tools_resource/static/js/toolresource.js (hunk 2 above)
Comment (shawncrawley) [inline hs_tools_resource/static/js/toolresource.js]: So you recommend adding a 'resource-title' class? Or should I make it an id?

Comment (mauriel133) [inline hs_tools_resource/static/js/toolresource.js]: An id would work.

**Comment by shawncrawley** [comment]
@mauriel133 Thanks for your suggestions. See if you can +1 now.

**Comment by mauriel133** [comment]
+1

**Pull request merged**